include/mbgl/storage/default_file_source.hpp

```diff
@@ -195,6 +190,11 @@ class DefaultFileSource : public FileSource {
      */
     void packDatabase(std::function<void(std::exception_ptr)> callback);
 
+    /*
+     * Enables or disables auto database file packing after deleteing regions and clearing ambient cache.
+     */
+    void setAutopack(bool);
```

Comment:
I think this API name is confusing. Naming this function `void runPackDatabaseAutomatically(bool)` would be a little clearer: the full reference to the related API is provided and it's obvious that if `true`, `packDatabase` is run. 

---

But why not implement this functionality so that `boolean runPackDatabase` is accepted as a variable in existing functions? e.g.:

> void deleteOfflineRegion(OfflineRegion&&, std::function<void(std::exception_ptr)>, bool runPackDatabase = true);

>  void clearAmbientCache(std::function<void (std::exception_ptr)>, bool runPackDatabase = true);

This way, the developer can decide with each function call whether to run `packDatabase()` automatically, and this new `"setAutopack"` API is not needed.

My understanding is that this wouldn't be a breaking change because if `runPackDatabase` is not passed in by the developer, it defaults to true which matches current behavior of both APIs. 

---

In the future, my preference would be to name all APIs around VACUUM/database defragmentation with those words explicitly. 

Unless `packDatabase()` does something additional to running `VACUUM`, what value do we provide the developer by wrapping a widely-known functionality (https://www.sqlitetutorial.net/sqlite-vacuum/) around this ambiguous word "pack"?

Reply:
> Unless packDatabase() does something additional to running VACUUM, what value do we provide the developer by wrapping a widely-known functionality (https://www.sqlitetutorial.net/sqlite-vacuum/) around this ambiguous word "pack"?

`VACUUM` is tied to sqlite, at some point we can switch to another DB provider. IMO it's better to keep API generic enough.

> But why not implement this functionality so that boolean runPackDatabase is accepted as a variable in existing functions?


This does not match well with the Android API: java does not support method parameters set by default.

Reply:
Also, we do not run `VACUUM` - we run `PRAGMA incremental_vacuum` which provides a different behavior, so we'd better not to disclose the implementation internals in the API.

Reply:
> VACUUM is tied to sqlite, at some point we can switch to another DB provider. IMO it's better to keep API generic enough.

> Also, we do not run VACUUM - we run PRAGMA incremental_vacuum which provides a different behavior, so we'd better not to disclose the implementation internals in the API.

Gotcha, makes sense to not reference VACUUM in the API name. Still, I think perhaps an alternative to "pack" should be considered for the future, unless there's a great reason why "pack" makes sense that I'm overlooking.

@pozdnyakov what do you think about specifying VACUUM / PRAGMA incremental_vacuum in the inline documentation though?

> This does not match well with the Android API: java does not support method parameters set by default.
 
😔 I guess our hands are tied here.

@pozdnyakov what do you think about changing the name from `setAutopack` to `runPackDatabaseAutomatically` then? (Also open to other suggestions.)

Finally -- is this function a global function for `OfflineManager`, so that if set, it would apply to every call to `deleteOfflineRegion` or `clearAmbientCache` that happens afterwards?

Reply:
> @pozdnyakov what do you think about specifying VACUUM / PRAGMA incremental_vacuum in the inline documentation though?

We could mention it as possible implementation detail, however, it's better IMO to just describe what it does - shrink the database file size, and point out the potential performance issues.

> @pozdnyakov what do you think about changing the name from setAutopack to runPackDatabaseAutomatically then? (Also open to other suggestions.)

`runPackDatabaseAutomatically` looks much better, thanks for proposing it!

> Finally -- is this function a global function for OfflineManager, so that if set, it would apply to every call to deleteOfflineRegion or clearAmbientCache that happens afterwards?

yeah, it'll apply to all the following calls of deleteOfflineRegion or clearAmbientCache.